File: backend/src/main/java/com/rahil/book_nexus/book/UnifiedSearchResponse.java
package com.rahil.book_nexus.book;

import com.rahil.book_nexus.external.GoogleBookSearchResult;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.List;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class UnifiedSearchResponse {

    private List<BookResponse> localBooks;
    private List<GoogleBookSearchResult> googleBooks;
    private String query;
    private int totalLocalResults;
    private int totalGoogleResults;

    public int getTotalResults() {
        return totalLocalResults + totalGoogleResults;
    }
}

File: frontend/src/app/services/models/UnifiedSearchResponse.ts
import { BookResponse } from './BookResponse';
import { GoogleBookSearchResult } from './GoogleBookSearchResult';

export interface UnifiedSearchResponse {
  localBooks: BookResponse[];
  googleBooks: GoogleBookSearchResult[];
  query: string;
  totalLocalResults: number;
  totalGoogleResults: number;
  totalResults: number;
}
